### ISSUE TITLE
React 18 (WebPack 5) doesn't work

### ISSUE THREAD

**Issue by vladles** [theKashey/used-styles#41]
Critical ERROR
Failed to compile.

```
Module not found: Error: Can't resolve 'fs' in '/Applications/MAMP/htdocs/charityrx-website/node_modules/scan-directory/dist'
ERROR in ./node_modules/scan-directory/dist/index.js 8:10-23
Module not found: Error: Can't resolve 'fs' in '/Applications/MAMP/htdocs/charityrx-website/node_modules/scan-directory/dist'

ERROR in ./node_modules/scan-directory/dist/index.js 12:12-27
Module not found: Error: Can't resolve 'path' in '/Applications/MAMP/htdocs/charityrx-website/node_modules/scan-directory/dist'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
        - install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "path": false }

ERROR in ./node_modules/used-styles/dist/es2015/reporters/critical.js 1:0-35
Module not found: Error: Can't resolve 'stream' in '/Applications/MAMP/htdocs/charityrx-website/node_modules/used-styles/dist/es2015/reporters'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }

ERROR in ./node_modules/used-styles/dist/es2015/reporters/used.js 1:0-35
Module not found: Error: Can't resolve 'stream' in '/Applications/MAMP/htdocs/charityrx-website/node_modules/used-styles/dist/es2015/reporters'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }

ERROR in ./node_modules/used-styles/dist/es2015/scanForStyles.js 2:0-30
Module not found: Error: Can't resolve 'fs' in '/Applications/MAMP/htdocs/charityrx-website/node_modules/used-styles/dist/es2015'

ERROR in ./node_modules/used-styles/dist/es2015/scanForStyles.js 3:0-47
Module not found: Error: Can't resolve 'path' in '/Applications/MAMP/htdocs/charityrx-website/node_modules/used-styles/dist/es2015'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
        - install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "path": false }

ERROR in ./node_modules/used-styles/dist/es2015/scanForStyles.js 4:0-33
Module not found: Error: Can't resolve 'util' in '/Applications/MAMP/htdocs/charityrx-website/node_modules/used-styles/dist/es2015'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
        - install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "util": false }

ERROR in ./node_modules/used-styles/node_modules/postcss/lib/input.js 6:35-50
Module not found: Error: Can't resolve 'path' in '/Applications/MAMP/htdocs/charityrx-website/node_modules/used-styles/node_modules/postcss/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
        - install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "path": false }

ERROR in ./node_modules/used-styles/node_modules/postcss/lib/map-generator.js 8:35-50
Module not found: Error: Can't resolve 'path' in '/Applications/MAMP/htdocs/charityrx-website/node_modules/used-styles/node_modules/postcss/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
        - install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "path": false }

ERROR in ./node_modules/used-styles/node_modules/postcss/lib/previous-map.js 8:35-50
Module not found: Error: Can't resolve 'path' in '/Applications/MAMP/htdocs/charityrx-website/node_modules/used-styles/node_modules/postcss/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
        - install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "path": false }

webpack compiled with 10 errors
```

**Comment by theKashey** [comment]
Vlad, with all respect, this is not how you can report an issue.
- there is no information what you did
- how you configuration look like
- why you think React 18 is related

------

So actually this issue is similar to https://github.com/theKashey/used-styles/issues/40 and the root cause that current implementation of `used-styles` __expects nodejs environment__ - like some packages exists and it can read files.

Long story short - if this error occurs at your SSR bundle (and used-styles cannot work for not SSR-cases by definition) - and you need to build that bundle with `node` as a compilation target - https://webpack.js.org/concepts/targets/#usage
